### PR TITLE
add visualization.shadertoy

### PR DESCRIPTION
**Pull request by un1versal**
lets see what happens upstream, ive pushed a few things to be reviewed, maybe that is ready for merging here?

### CODE REVIEW

**Comment by wsnipex** [comment]
I guess we need a unified upstream that includes support for all platforms.
@notspiff since your the original author this is also a viz, would make sense to use yours as well, if you're up for merging stuff together:
popcornmix' fork is ahead, but lacks the debian stuff from topfs2

**Comment by notspiff** [comment]
i can consolidate it, sure, but give me a day, i've done too much non-work things today already :)

**Comment by wsnipex** [comment]
heh, no hurry. Much appreciated!

**Comment by notspiff** [comment]
okay, i had to wait for another simulation.

i merged popcornmix/master, afedchin/dx11, topfs2/master into my fork, resolved some minors (gles one had not been tested with gl).

i cannot guarantee that i got everything right merging the shaders. there were conflicting changes in gles and dx11. i tried to choose dx11 when that made sense since it was mainly cosmetics (which likely are important for the compiler).

for audiovisual.glsl.frag i chose the gles one as it looked less hacky than the dx11 changes.

it builds and runs for me (opengl), but i cannot guarantee that everything is optimal for gles / dx11.

**Comment by wsnipex** [comment]
:)
@un1versal please change to notpiffs repo

**Comment by un1versal** [comment]
So this means I should send PRs to notspiff also that I have waiting at topfs2 right?

give me a sec

**Comment by notspiff** [comment]
yes, it will make me upstream.

**Comment by un1versal** [comment]
@wsnipex done, notspiff master needs a twice over I submitted some fixes and the pr's from topfs2 already but I suspect a few things were lost

Ill do what I can but a lot may be over me.

edit done what I could see.

File: visualization.shadertoy/platforms.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+all
```
Comment (wsnipex) [inline visualization.shadertoy/platforms.txt]: glew depends for ios are missing in shadertoy
@notspiff is it supposed to work there?

Comment (notspiff) [inline visualization.shadertoy/platforms.txt]: Glew is opengl. Ios is gles, so no.
29. okt. 2015 10:49 skrev "Wolfgang Schupp" notifications@github.com:

Comment (un1versal) [inline visualization.shadertoy/platforms.txt]: I wasnt sure if this was all, or `linux osx windows rbpi` only Im not sure what the valid entries are supposed to be here.

I added `all`` due comment by wsnipex at https://github.com/xbmc/repo-binary-addons/pull/12#issuecomment-151862230

Comment (wsnipex) [inline visualization.shadertoy/platforms.txt]: well, that was meant as all supported platform. so please:
linux osx windows rbpi

Comment (un1versal) [inline visualization.shadertoy/platforms.txt]: Yes sorry, got confused with what was supported vs topfs2, fixed now.

**Comment by wsnipex** [comment]
thanks!

File: visualization.shadertoy/screensaver.shadertoy.txt [hunk 2]
```diff
@@ -0,0 +1 @@
+visualization.shadertoy https://github.com/notspiff/visualization.shadertoy master
```
Comment (stefansaraev) [inline visualization.shadertoy/screensaver.shadertoy.txt:1]: visualization.shadertoy/visualization.shadertoy.txt 
because it is not a screensaver. is it ?

Comment (un1versal) [inline visualization.shadertoy/screensaver.shadertoy.txt:1]: well spoted, c&p strikes again fixed in https://github.com/xbmc/repo-binary-addons/pull/15

**Pull request merged**